src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -460,15 +465,21 @@ private void doData(
         int length,
         Flyweight extension)
     {
-        final DataFW data = dataRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+        final DataFW.Builder builder = dataRW.wrap(writeBuffer, 0, writeBuffer.capacity())
             .routeId(routeId)
             .streamId(replyId)
             .traceId(traceId)
             .authorization(authorization)
             .flags(flags)
             .budgetId(budgetId)
-            .reserved(reserved)
-            .payload(buffer, index, length)
+            .reserved(reserved);
+
+        if (buffer != null)
+        {
+            builder.payload(buffer, index, length);
+        }
```

Comment:
Can `doData` be refactored to receive `payload` parameter as type `Flyweight` instead of separate `buffer`, `offset` and `limit`, making it easy to pass `null` when needed without the `null` checks only required because we're trying to split it into `buffer`, `offset` and `length`?

Reply:
Perhaps change this to:
```
builder.payload(payload != null ? octetsRO.wrap(payload.buffer(), payload.offset(), payload.length()) : null);
```

Then we can go back to the style of code before these changes, where the builder did not need to be extracted separately.

As an aside, perhaps we should consider adding `Flyweight.Builder.wrap(flyweight)` and `Flyweight.Builder.tryWrap(flyweight)`, where `tryWrap` returns `null` if `flyweight` is `null`, then the code here would become:
```
builder.payload(octetsRO.tryWrap(payload));
```

Alternatively, we could generate an overload of `DataFW.Builder.payload(Flyweight payload)` that has the same implementation as `DataFW.Builder.payload(OctetsFW payload)`, then the code would become:
```
builder.payload(payload);
```
This last option seems best to me, what do you think?